File: documents-service-api/src/main/kotlin/documents/service/api/DocumentsOperations.java
package documents.service.api;

import documents.service.api.queries.finddocuments.FindDocumentsQuery;
import documents.service.api.queries.finddocuments.FindDocumentsResult;
import io.micronaut.http.annotation.Body;
import io.micronaut.http.annotation.Post;

import javax.validation.constraints.NotNull;

public interface DocumentsOperations {

    @Post
    FindDocumentsResult find(@Body @NotNull FindDocumentsQuery query);

}

File: documents-service-api/src/main/kotlin/documents/service/api/DocumentsOperations.kt
package documents.service.api

import documents.service.api.queries.finddocuments.FindDocumentsResult
import io.micronaut.http.annotation.Get
import org.jetbrains.annotations.NotNull

interface DocumentsOperations {

    @Get("/{policyNumber}")
    fun find(@NotNull policyNumber: String): FindDocumentsResult

}
